File: programmers/pccp/wonhyeok/유전법칙.js
function getGene(pose) {
    let [n, p] = pose
    let stack = []
    
    p--
    while (n>1) {
        stack.push(p%4)
        n--
        p = parseInt(p/4)
    }
    
    while (stack.length > 0) {
        num = stack.pop()
        if (num == 0) return 'RR'
        if (num == 3) return 'rr'
    }
    return 'Rr'
}
 
function solution(queries) {
    return queries.map(getGene)
}

File: programmers/pccp/wonhyeok/유전법칙.py
def get_gene(pose):
    n, p = pose
    if n == 1: return 'Rr'
    
    mother_gene = get_gene((n-1, (p-1)//4+1))
    case = ('RR','Rr','Rr','rr')
    
    if mother_gene == 'Rr': return case[p%4-1]
    else: return mother_gene
 
def solution(queries):
    return [*map(get_gene, queries)]
